ui/app/components/variable-form.js

```diff
@@ -20,6 +20,9 @@ const EMPTY_KV = {
   warnings: EmberObject.create(),
 };
 
+// Capture characters that are not _, letters, or numbers using Unicode.
+const invalidKeyCharactersRegex = new RegExp(/[^_\p{Letter}\p{Number}]/gu);
```

Comment:
`praise`: Beautiful job writing this regular expression. And the tests help with documentation of whats happening here, too.

Reply:
Thanks AI 😄